src/platform/packages/shared/kbn-data-grid-in-table-search/src/matches/use_find_matches.tsx

```diff
@@ -0,0 +1,244 @@
+          columns: visibleColumns,
+          onScrollToActiveMatch,
+        });
+      }
+
+      // const duration = window.performance.now() - startTime;
+      // console.log(duration);
+    };
+
+    const RenderCellsShadowPortal: UseFindMatchesState['renderCellsShadowPortal'] = () => (
+      <AllCellsMatchesCounter
+        key={numberOfRuns}
+        inTableSearchTerm={inTableSearchTerm}
+        rowsCount={rows.length}
+        visibleColumns={visibleColumns}
+        renderCellValue={renderCellValue}
+        onFinish={onFinish}
+      />
+    );
+
+    setState((prevState) => ({
+      ...prevState,
+      isProcessing: true,
+      renderCellsShadowPortal: RenderCellsShadowPortal,
+    }));
+  }, [setState, renderCellValue, visibleColumns, rows, inTableSearchTerm, onScrollToActiveMatch]);
+
+  const goToPrevMatch = useCallback(() => {
+    setState((prevState) => changeActiveMatchInState(prevState, 'prev', onScrollToActiveMatch));
+  }, [setState, onScrollToActiveMatch]);
+
+  const goToNextMatch = useCallback(() => {
+    setState((prevState) => changeActiveMatchInState(prevState, 'next', onScrollToActiveMatch));
+  }, [setState, onScrollToActiveMatch]);
+
+  const resetState = useCallback(() => {
+    setState(INITIAL_STATE);
+  }, [setState]);
+
+  return useMemo(
+    () => ({
+      matchesCount,
+      activeMatchPosition,
+      goToPrevMatch,
+      goToNextMatch,
+      resetState,
+      isProcessing,
+      renderCellsShadowPortal,
+    }),
+    [
+      matchesCount,
+      activeMatchPosition,
+      goToPrevMatch,
+      goToNextMatch,
+      resetState,
+      isProcessing,
+      renderCellsShadowPortal,
+    ]
+  );
+};
+
+function getActiveMatchForPosition({
+  matchPosition,
+  matchesList,
+  columns,
+}: {
+  matchPosition: number;
+  matchesList: RowMatches[];
+  columns: string[];
+}): ActiveMatch | null {
+  let traversedMatchesCount = 0;
+
+  for (const rowMatch of matchesList) {
+    const rowIndex = rowMatch.rowIndex;
+
+    if (traversedMatchesCount + rowMatch.rowMatchesCount < matchPosition) {
+      // going faster to next row
+      traversedMatchesCount += rowMatch.rowMatchesCount;
+      continue;
+    }
+
+    const matchesCountPerColumnId = rowMatch.matchesCountPerColumnId;
+
+    for (const columnId of columns) {
+      // going slow to next cell within the row
+      const matchesCountInCell = matchesCountPerColumnId[columnId] ?? 0;
+
+      if (
+        traversedMatchesCount < matchPosition &&
+        traversedMatchesCount + matchesCountInCell >= matchPosition
+      ) {
+        // can even go slower to next match within the cell
+        return {
+          rowIndex: Number(rowIndex),
+          columnId,
+          matchIndexWithinCell: matchPosition - traversedMatchesCount - 1,
+        };
+      }
+
+      traversedMatchesCount += matchesCountInCell;
+    }
+  }
+
+  // no match found for the requested position
+  return null;
+}
+
+let prevJumpTimer: NodeJS.Timeout | null = null;
```

Comment:
Could sharing this reference potentially cause issues with multiple searchable grids on screen at once like in dashboards? Doesn't really seem like it from the code, but wanted to raise it regardless.

Reply:
Potentially yes, it would cancel a jump inside one panel when a jump is triggered in another one. Not likely in reality as changing between panels would take longer when the jump itself. 

Having the timeout helps with a flow when users presses prev/next buttons or the shortcut too quickly inside one panel. Less visual noise with the highlights.

Reply:
Makes sense, thanks for confirming. Agree it doesn't seem like it would actually happen, and we can revisit if we find some users stepping through results at light speed 😄